src/schema/credit_card.js

```diff
@@ -61,6 +61,30 @@ const CreditCardType = new GraphQLObjectType({
       type: GraphQLNonNull(GraphQLInt),
       description: "Credit card's expiration year",
     },
+    street1: {
+      type: GraphQLNonNull(GraphQLString),
+      description: "Billing address street1",
+    },
+    street2: {
+      type: GraphQLString,
+      description: "Billing address street2",
+    },
+    city: {
+      type: GraphQLNonNull(GraphQLString),
+      description: "Billing address city",
+    },
+    state: {
+      type: GraphQLNonNull(GraphQLString),
+      description: "Billing address state",
+    },
+    country: {
+      type: GraphQLNonNull(GraphQLString), // TODO: We may make this type more strict by throwing ISO "ALPHA-2 Codes
+      description: "Billing address country code",
+    },
+    postal_code: {
+      type: GraphQLNonNull(GraphQLString),
+      description: "Billing address postal code",
+    },
```

Comment:
So this type basically just passes the specified fields from Gravity down to the client. I wonder if we should use more appropriate names or leave them as they are now.

Reply:
Maybe grouping these under an `address` field would be nice? Maybe it can even share an ‘address’ type with the fulfilment types, which means you could theoretically re-use components as well.

Reply:
This is by no means a requirement, just thinking out loud because you were musing about it too.

Reply:
Noticing that the name `state` is very confusing in React...

Reply:
:ceiling-cat:

Heh, but a "federal state" is actually a [thing](https://en.wikipedia.org/wiki/Federation).

When "state" isn't good enough, the generic & jargony name for that kind of entity is "first-level administrative division" — which encompasses states, provinces, cantons, etc — and is often abbreviated in code as `admin1`.

Reply:
TIL. Well stated!